### PR TITLE
feat: Add action for deploying a plugin

### PR DESCRIPTION
**Pull request by SilviaZeta**
## Description
Add new action for deploying a plugin.

## Checklist:

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request). If you're unsure about any of these, don't hesitate to leave a comment on this pull request!

- [ ] I have read the gliff.ai Contribution Guide.
- [x] I have requested to **pull a branch** and not from main.
- [ ] I have checked all commit message styles match the requested structure.
- [ ] My code follows the style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have performed a self-review of my own code.
- [x] I have assigned **3 or less** reviewers.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.
- [ ] If appropriate, I have bumped any version numbers.

### CODE REVIEW

File: .github/workflows/plugin-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: Deploy plugin to Azure function app
+
+on:
+  workflow_call:
+    secrets:
```
Comment (ChasNelson1990) [inline .github/workflows/plugin-deploy.yml]: I would not have added this `secrets` block myself. Can you explain why it's here? (Trying to work out if it's something I _should_ be  doing lol)

File: .github/workflows/plugin-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          token: ${{ secrets.GIT_TOKEN }}
+      - name: "Login via Azure CLI"
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: setup Python ${{ secrets.PYTHON_VERSION }} environment
```
Comment (ChasNelson1990) [inline .github/workflows/plugin-deploy.yml]: Should we not hardcode the py version as we want it to be a specific version for all plugins at this stage?

Comment (SilviaZeta) [reply .github/workflows/plugin-deploy.yml]: what do you mean by "hardcode" here? I made the python version a secret precisely so we can have a different version for each plugin, if we need to.

File: .github/workflows/plugin-deploy.yml (hunk 1 above)
Comment (SilviaZeta) [reply .github/workflows/plugin-deploy.yml]: Ah, I had to add that block to define which secrets are passed to the reusable action when this is called from some other repo. Without that block the secrets just weren't available.

**Review by philipjackson — APPROVED** [review]

File: .github/workflows/plugin-deploy.yml (hunk 1 above)
Comment (ChasNelson1990) [reply .github/workflows/plugin-deploy.yml]: I think some of these should be inputs, not secrets. See: https://github.com/gliff-ai/.github/blob/e9fab1e3f6b81c5be9af93c24e0a4725c6310e4b/.github/workflows/dependency-updates.yml#L4

**Review by ChasNelson1990 — CHANGES_REQUESTED** [review]

File: .github/workflows/plugin-deploy.yml (hunk 1 above)
Comment (SilviaZeta) [reply .github/workflows/plugin-deploy.yml]: oh, I didn't know of "inputs".. makes sense, given that some of the vars are not real secrets, but just data passed to the action.. I liked to the idea of changing a plugin without having to change the code, but possibly not what we want given that changes in secrets are not versioned.

**Review by ChasNelson1990 — APPROVED** [review]

**Pull request merged**